lib/clouds/shared_functions.py

```diff
@@ -1187,33 +1021,31 @@ def base_check_images(self, vmc_name, vm_templates, registered_imageid_list, map
     @trace
     def populate_cloudconfig(self, obj_attr_list) :
         '''
-        CloudBench should be passing us a more complex object for userdata,
-        but is only passing us a script instead. So, we have to wrap the
-        userdata in formal cloud-config syntax in order to be able to use
-        if with cloud images that have cloud-init configured correctly.
+        TBD
         '''
-        if ("userdata" not in obj_attr_list or not obj_attr_list["userdata"]) and obj_attr_list["use_vpn_ip"].lower() == "false" :
+        if ("userdata" not in obj_attr_list or str(obj_attr_list["userdata"]).lower() == "false") and obj_attr_list["use_vpn_ip"].lower() == "false" :
             #cbdebug("Skipping userdata: " + str(obj_attr_list["userdata"]), True)
             return False
 
-        cloudconfig = """
-#cloud-config
-write_files:"""
-        if "userdata" in obj_attr_list and obj_attr_list["userdata"] :
-            cloudconfig += """
-  - path: /tmp/userscript.sh
-    content: |
-"""
-            for line in obj_attr_list["userdata"].split("\n")[:-1] :
-                cloudconfig += "      " + line + "\n"
-
-        # We need the VPN's IP address in advance, which was solved before
-        # the previous VPN support was gutted, but since we're left to do it
-        # on our own, we need cloud-config to send our VPN configuration file
-        # in advance.
-        conf_destination = "/etc/openvpn/" + obj_attr_list["cloud_name"] + "_client-cb-openvpn-cloud.conf"
+        cloudconfig = "#cloud-config\n"
```

Comment:
Please bring back the multi-line string syntax. It makes understanding cloud-config much easier.

If you were not aware in python, multi-line strings can be appended to `on-the-fly` in python by simply concatenating regular strings to multi-line strings within the same syntax.

Reply:
yeah, ok... I actually like the single-line string syntax (this is just my personal preference), but you object and requires multi-line syntax, I will bring those back, sure.